### PR TITLE
 [sourcekitd] Capture diagnostics with invalid locations for compile notifications

### PR DESCRIPTION
**Pull request by benlangmuir**
Stop filtering out diagnostics with invalid locations in the editor diagnostic consumer, and instead capture them separately so that we can include them in did-compile notifications.

rdar://39225000

### CODE REVIEW

**Comment by benlangmuir** [comment]
@swift-ci please smoke test

File: tools/SourceKit/lib/SwiftLang/SwiftEditor.cpp [hunk 1]
```diff
@@ -138,24 +140,29 @@ void EditorDiagConsumer::handleDiagnostic(
...
-        SM.findBufferContainingLoc(F.getRange().getStart()) != BufferID)
+    if (F.getRange().isInvalid() || !BufferID ||
+        SM.findBufferContainingLoc(F.getRange().getStart()) != *BufferID)
       continue;
     unsigned Offset = SM.getLocOffsetInBuffer(F.getRange().getStart(),
-                                              BufferID);
+                                              *BufferID);
     unsigned Length = F.getRange().getByteLength();
     SKInfo.Fixits.push_back({ Offset, Length, F.getText() });
   }
```
Comment (nathawes) [inline tools/SourceKit/lib/SwiftLang/SwiftEditor.cpp]: I'm not sure this makes much difference, so up to you, but I thought the above might be more readable with a top-level `if (BufferId) { check for non-input buffer, set filename + offset + ranges + fixits } else { set unknown filename }`

**Review by nathawes — APPROVED** [review]
LGTM!

File: tools/SourceKit/lib/SwiftLang/SwiftEditor.cpp (hunk 1 above)
Comment (benlangmuir) [reply tools/SourceKit/lib/SwiftLang/SwiftEditor.cpp]: Ah, good point.  I'll do that.

**Comment by benlangmuir** [comment]
@swift-ci please smoke test

**Pull request merged**